core/src/main/java/io/lionweb/lioncore/java/language/LionCoreBuiltins.java

```diff
@@ -1,27 +1,36 @@
 package io.lionweb.lioncore.java.language;
 
-import io.lionweb.lioncore.java.serialization.JsonSerialization;
+import io.lionweb.lioncore.java.LionWebVersion;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import javax.annotation.Nonnull;
 
 public class LionCoreBuiltins extends Language {
-  private static final LionCoreBuiltins INSTANCE = new LionCoreBuiltins();
+  // We may have one instance of this class per LionWeb version, and we initialize
+  // them lazily
+  private static final Map<LionWebVersion, LionCoreBuiltins> INSTANCES = new HashMap<>();
 
   /** This is private to prevent instantiation and enforce the Singleton pattern. */
-  private LionCoreBuiltins() {
-    super("LionCore_builtins");
+  private LionCoreBuiltins(@Nonnull LionWebVersion lionWebVersion) {
+    super(lionWebVersion, "LionCore_builtins");
     setID("LionCore-builtins");
     setKey("LionCore-builtins");
-    setVersion(JsonSerialization.DEFAULT_SERIALIZATION_FORMAT);
-    PrimitiveType string = new PrimitiveType(this, "String");
-    new PrimitiveType(this, "Boolean");
-    new PrimitiveType(this, "Integer");
-    new PrimitiveType(this, "JSON");
+    setVersion(lionWebVersion.getValue());
+    PrimitiveType string = new PrimitiveType(lionWebVersion, this, "String");
+    new PrimitiveType(lionWebVersion, this, "Boolean");
+    new PrimitiveType(lionWebVersion, this, "Integer");
+    if (lionWebVersion.equals(LionWebVersion.v2023_1)) {
```

Comment:
This kind of compatibility will be pretty hard to maintain on the long term.

Reply:
I am not sure this would be the case as I am not sure how many versions we will have and how much they will vary from each other. For now, it seems the simplest approach. Or do you think we should refactor it?